core/src/main/java/gay/oss/gatos/core/collections/BaseCollection.java

```diff
@@ -47,12 +57,138 @@ public void insert(T obj) {
+        }
+        return list;
+    }
+
+    /**
+     * Updates a document. Only non-null fields will be updated.
+     *
+     * @param id  The ID of the document to update.
+     * @param obj The POJO to update with.
+     * @return The updated POJO.
+     */
+    public T update(UUID id, T obj) {
+        List<Bson> updates = getNonNullUpdates(obj);
+        if (!updates.isEmpty()) {
+            getCollection().updateOne(idFilter(id), Updates.combine(updates));
+        }
+        return get(id);
+    }
+
+    /**
+     * Deletes a document.
+     *
+     * @param id The ID of the document to delete.
+     */
+    public void delete(UUID id) {
+        getCollection().deleteOne(idFilter(id));
+    }
+
+    /**
+     * Creates an ID filter.
+     *
+     * @param id The ID to filter by.
+     * @return The filter.
+     */
+    private static Bson idFilter(UUID id) {
+        return eq("_id", id);
+    }
+
+    /**
+     * Creates a list of non-null updates from a POJO.
+     *
+     * @param obj The POJO.
+     * @return A list of updates.
+     */
+    private static List<Bson> getNonNullUpdates(Object obj) {
+        return createPropertyDescriptorStream(obj.getClass())
+            .filter(BaseCollection::hasGetter)
+            .map(descriptor -> getField(descriptor, obj))
+            .filter(Objects::nonNull)
+            .map(Field::toUpdate)
+            .toList();
+    }
+
+    /**
+     * Creates a stream of {@code PropertyDescriptors} from a class. The {@link Object} class is excluded.
+     *
+     * @param clazz The class.
+     * @return A stream of {@code PropertyDescriptors}.
+     */
+    private static Stream<PropertyDescriptor> createPropertyDescriptorStream(Class<?> clazz) {
+        try {
+            return Stream.of(Introspector.getBeanInfo(clazz, Object.class).getPropertyDescriptors());
+        } catch (IntrospectionException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Checks if a {@code PropertyDescriptor} has a getter.
+     *
+     * @param descriptor The {@code PropertyDescriptor}.
+     * @return {@code true} if the {@code PropertyDescriptor} has a getter, {@code false} otherwise.
+     */
+    private static boolean hasGetter(PropertyDescriptor descriptor) {
+        return descriptor.getReadMethod() != null;
+    }
+
+    /**
+     * Creates a {@code Field} object by invoking the getter of a {@code PropertyDescriptor} on the passed object.
+     *
+     * @param descriptor The {@code PropertyDescriptor}.
+     * @param obj        The object to invoke the getter on.
+     * @return A {@code Field} object.
+     */
+    @Nullable
+    private static Field getField(PropertyDescriptor descriptor, Object obj) {
+        try {
+            Object value = descriptor.getReadMethod().invoke(obj);
+            if (value != null) {
+                return new Field(descriptor.getName(), value);
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
This is fine for now, but we should set up slf4j or something for future

Reply:
I can do that rn, what implementation do we want